Shopify/cli
# Add client-id and reset as global flags

### WHY are these changes introduced?

Standardizes the handling of `client-id` and `reset` flags across all app commands by moving them to the shared app flags.

### WHAT is this pull request doing?

- Moves `client-id` and `reset` flags to shared app flags
- Updates all commands to use the shared flag implementations
- Standardizes the usage of `client-id` and `reset` flags in `linkedAppContext` calls

### How to test your changes?

1. Run various app commands with `--client-id` flag
2. Run various app commands with `--reset` flag
3. Verify that both flags work consistently across all commands

### Checklist
